### PR TITLE
Add pyb2d package

### PR DESCRIPTION
**Pull request by DerThorsten**
### Description
 package for https://github.com/pyb2d/pyb2d/ which is a pybind11 based Box2D / liquidfun python wrapper.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation (does not need new documentation in pyodide since its just a package)

### CODE REVIEW

**Comment by DerThorsten** [comment]
the 'ci/circleci: test-js' test seems to fail unrelated to my changes.
```
../../build/pyproxy.gen.d.ts:249:27
  ✖  249:27  Cannot find name bool.  
  ✖  250:25  Cannot find name bool.  

  2 errors
```

Other PRs also fail with that: https://app.circleci.com/pipelines/github/ryanking13/pyodide/34/workflows/affba5ae-ab35-4ef5-8af6-a2cd32f0a228/jobs/490

File: packages/pyb2d/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  cflags: -Wno-implicit-function-declaration
+
+requirements:
+  run:
+    - numpy
+
+test:
+  imports:
+    - b2d
+    - b2d.testbed
```
Comment (hoodmane) [inline packages/pyb2d/meta.yaml:21]:
```suggestion
    - b2d.testbed

```
Please add a newline at end of file.

File: packages/pyb2d/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  version: 0.4.4
+
+source:
+  url: https://github.com/pyb2d/pyb2d/archive/refs/tags/0.4.4.tar.gz
+  sha256: 9e8d613bef2a5e0dc25da96ef23e5a3bf9f07eac9e07dfa632fee94df2a5273c
+
+
+build:
+  script: python -m pip install pybind11
+  cflags: -Wno-implicit-function-declaration
```
Comment (hoodmane) [inline packages/pyb2d/meta.yaml:12]: Why `-Wno-implicit-function-declaration`?

Comment (DerThorsten) [reply packages/pyb2d/meta.yaml]: I believe this is a left over from some debugging. Just pushed a version without that flag which hopefully build.

**Comment by DerThorsten** [comment]
@hoodmane works now also without the questionable `-Wno-implicit-function-declaration`

File: docs/project/changelog.md [hunk 3]
```diff
@@ -125,6 +125,7 @@ substitutions:
   {pr}`1883`
 
 - New packages: `logbook`
+- New packages: `pyb2d`
```
Comment (hoodmane) [inline docs/project/changelog.md:128]:
```suggestion
- New packages: `logbook`, `pyb2d`
```

**Review by hoodmane — APPROVED** [review]

**Pull request merged**